src/redux/missions/missionsSlice.js

```diff
@@ -0,0 +1,48 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { MISSIONS_URL } from '../../config';
+import getData from '../../api';
+
+const initialState = {
+  missions: [],
+  isLoading: true,
+};
+
+export const fetchMissions = createAsyncThunk('missions/fetchMissions', async () => {
+  const response = await getData(MISSIONS_URL);
+  return response;
+});
```

Comment:
- Kindly, you are storing the entire data fetched from the API to redux, this may result in unnecessary overhead and slow down the application, 👌 it would be better to store the necessary data only,  you are using only `mission_id` `mission_name` and `descriptions ` so store those fields only it to redux store 👍👍
<img width="274" alt="Screenshot 2023-03-09 at 5 55 31 PM" src="https://user-images.githubusercontent.com/52010789/224063167-fcc4d462-6f67-413f-b9bc-bbf2306dfea5.png">

Reply:
@binhussen that is not accurate at all. If you look at the getData function located in the api.js file, you will see that only the mission_id, mission_name, and description data is stored not the whole data. [Have a look](https://github.com/Timbar09/React-Redux-Group-Project-Space-Travelers-Hub/pull/32/files#diff-d45b1674ab37689f6ca6dc634fe29959c463fc30cb46746c3b187de88432ae4fR5).